rpc/clients/typescript/test/ws_client_test.ts

```diff
@@ -1,507 +1,434 @@
+                    await exchange.cancelOrder(order).awaitTransactionSuccessAsync({ from: makerAddress });
+
+                    // Subscribe to order events and assert that only a single cancel event was received.
+                    const now = new Date(Date.now()).getTime();
+                    await deployment.client.subscribeToOrdersAsync((orderEvents: OrderEvent[]) => {
+                        // Ensure that the correct cancel event was logged.
+                        expect(orderEvents.length).to.be.eq(1);
+                        const [orderEvent] = orderEvents;
+                        expect(orderEvent.endState).to.be.eq(OrderEventEndState.Cancelled);
+                        expect(orderEvent.fillableTakerAssetAmount).to.be.bignumber.eq(constants.ZERO_AMOUNT);
+                        expect(orderEvent.signedOrder).to.be.deep.eq(order);
+                        assertRoughlyEquals(orderEvent.timestampMs, now, twoSecondsInMs);
+                        expect(orderEvent.contractEvents.length).to.be.eq(1);
+
+                        // Ensure that the contract event is correct.
+                        const [contractEvent] = orderEvent.contractEvents;
+                        expect(contractEvent.address).to.be.eq(exchangeAddress);
+                        expect(contractEvent.kind).to.be.equal(ContractEventKind.ExchangeCancelEvent);
+                        expect(contractEvent.logIndex).to.be.eq(0);
+                        expect(contractEvent.isRemoved).to.be.false();
+                        expect(contractEvent.txIndex).to.be.eq(0);
+                        const hashLength = 66;
+                        expect(contractEvent.blockHash.length).to.be.eq(hashLength);
+                        expect(contractEvent.blockHash).to.not.be.eq(constants.NULL_BYTES32);
+                        expect(contractEvent.txHash.length).to.be.eq(hashLength);
+                        expect(contractEvent.txHash.length).to.be.eq(hashLength);
+                        const parameters = contractEvent.parameters as ExchangeCancelEvent;
+                        parameters.makerAddress = parameters.makerAddress.toLowerCase();
+                        parameters.senderAddress = parameters.makerAddress;
+                        expect(parameters.feeRecipientAddress.toLowerCase()).to.be.eq(order.feeRecipientAddress);
+                        expect(parameters.makerAddress.toLowerCase()).to.be.eq(makerAddress);
+                        expect(parameters.makerAssetData).to.be.eq(order.makerAssetData);
+                        expect(parameters.orderHash).to.be.eq(orderHashUtils.getOrderHashHex(order));
+                        expect(parameters.senderAddress.toLowerCase()).to.be.eq(makerAddress);
+                        expect(parameters.takerAssetData).to.be.eq(order.takerAssetData);
+                        done();
+                    });
+                })().catch(done);
+            });
```

Comment:
Let's add a test where we create a subscription, and then unsubscribe with the `subscriptionID` received upon subscription.